pubtools/_pulp/tasks/delete.py

```diff
@@ -0,0 +1,709 @@
+from pubtools.pulplib import (
+    Criteria,
+    Matcher,
+    RpmUnit,
+    ModulemdUnit,
+    FileUnit,
+    ErratumUnit,
+)
+
+from pubtools._pulp.services.pulp import PulpClientService
+from pubtools._pulp.services.collector import CollectorService
+from pubtools._pulp.task import PulpTask
+from pubtools._pulp.tasks.common import Publisher
+from pubtools._pulp.arguments import SplitAndExtend
+
+LOG = logging.getLogger("pubtools.pulp")
+
+step = PulpTask.step
+
+MODULEMD_REGEX = re.compile(r"^[-.+\w]+:[-.+\w]+:\d+(:[-.+\w]+){0,2}$")
+
+
+@attr.s
+class ClearedRepo(object):
+    """Represents a single repo where contents were removed."""
+
+    tasks = attr.ib()
+    """The completed Pulp tasks for removing content from this repo."""
+
+    repo = attr.ib()
+    """The repo which where content was cleared."""
+
+    content = attr.ib()
+    """The content that was cleared from the repo"""
+
+
+@attr.s
+class RemoveUnitItem(object):
+    """Represents a unit with destination repos"""
+
+    unit = attr.ib()
+    """A Pulp unit"""
+
+    repos = attr.ib()
+    """The list of repos from where the unit to be removed"""
+
+
+class Delete(PulpClientService, CollectorService, Publisher, PulpTask):
+    def __init__(self, *args, **kwargs):
+        super(Delete, self).__init__(*args, **kwargs)
+        self.to_await = []
+
+    def add_args(self):
+        super(Delete, self).add_args()
+
+        self.add_publisher_args(self.parser)
+
+        self.parser.add_argument(
+            "--repo",
+            help="remove content from these comma-seperated repositories ",
+            type=str,
+            action=SplitAndExtend,
+            split_on=",",
+        )
+
+        self.parser.add_argument(
+            "--advisory",
+            help="remove packages in the advisory",
+            type=str,
+        )
+
+        self.parser.add_argument(
+            "--file",
+            help="remove these comma-separated rpm, srpm, modulemd or iso file(s)",
+            type=str,
+            action=SplitAndExtend,
+            split_on=",",
+        )
+
+        self.parser.add_argument(
+            "--signing-key",
+            help="remove the content with these signing-keys(s)",
+            type=str,
+            action=SplitAndExtend,
+            split_on=",",
+        )
+
+    def fail(self, *args, **kwargs):
+        LOG.error(*args, **kwargs)
+        sys.exit(30)
+
+    def run(self):
+        repos = {}
+        rpm_f = f_return()
+        file_f = f_return()
+        module_f = f_return()
+        adv_f = f_return()
+        signing_keys = self.args.signing_key
+        repo_names = self.args.repo
+        advisory = self.args.advisory
+
+        if not (self.args.file or self.args.advisory):
+            self.fail("One of --file or --advisory is required")
+
+        # separates files into rpms, isos and modules
+        rpms, files, modules, _ = self.separate_files(self.args.file or [])
+
+        # collect skipped items
+        # TODO
```

Comment:
What does this mean actually?

Reply:
There are some files that are skipped or not found during the process. Should we send those files to collector or logging them is enough?
This is a TODO to collect those skipped files.

Reply:
That's rather open-ended I'd say.  I don't think much thought was put into the semantics of "skipped" in pub originally and I don't think we have to stick with it.  I'd say that it's OK to err on the side of less work / less complexity, and just leave it out.